### PR TITLE
AI Assistant: compute feature availability optimistically

### PR DESCRIPTION
**Pull request by retrofox**
When the site archives the requests limit, depending on the current plan/tier,m we should compute if the limit has been achieved, and if the site requires an upgrade. This PR takes over the `isOverLimit` and `requiredUpgrade` props, based on the server-side computing. 

For now, it works with the default plan.

Fixes #

## Proposed changes:

* AI Assistant: compute feature availability optimistically

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Test with a site with a Free plan
* Go to the block editor
* Open the Jetpack AI side
* Identify the Usage panel
* Open the Redux dev tool
* Identify the `wordpress-com/plans` store
* Open the dev console
* Let's store the testing state. Copy the following code and paste it into your dev console:

```js
wp.data.dispatch( 'wordpress-com/plans' ).storeAiAssistantFeature( {
    "hasFeature": false,
    "isOverLimit": false,
    "requestsCount": 15,
    "requestsLimit": 20,
    "requireUpgrade": false,
    "upgradeType": "default",
    "usagePeriod": {
        "currentStart": "2023-11-06 00:00:00",
        "nextStart": "2023-12-06 00:00:00",
        "requestsCount": 15
    },
    "currentTier": {
        "value": 1
    },
    "nextTier": {
        "slug": "ai-assistant-tier-100",
        "limit": 100,
        "value": 100
    }
} )
```
* Confirm the Usage panel reflects the right stored
* Increase the requests counter by dispatching the action:

```js
wp.data.dispatch( 'wordpress-com/plans' ).increaseAiAssistantRequestsCount()
```
* Confirm the request counter increases
* Confirm the panel changes the bar color and info once the limit is achieved

https://github.com/Automattic/jetpack/assets/77539/7d87d230-3ce2-41f5-af8c-311476b03a6e

### CODE REVIEW

File: projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const INITIAL_STATE: PlanStateProps = {
 			hasFeature: false,
 			isOverLimit: false,
 			requestsCount: 0,
-			requestsLimit: 0,
+			requestsLimit: 1000,
```
Comment (CGastrell) [inline projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts:19]: I think we can settle for 20 on this one as it should be the free default value?

Comment (CGastrell) [reply projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts:19]: or zero if we want to be restrictive with our defaults

Comment (retrofox) [reply projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts:19]: yes. Let me update the code

**Review by CGastrell — APPROVED** [review]
Left a comment, but works nicely! :shipit:

**Pull request merged**